src/metrics.rs

```diff
@@ -0,0 +1,38 @@
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::Arc;
+
+#[derive(Debug, Clone)]
+pub struct Counter {
+    pub value: Arc<AtomicU64>,
+}
+
+impl Counter {
+    pub fn get(&self) -> u64 {
+        self.value.load(Ordering::Relaxed)
+    }
+
+    pub fn inc(&self) -> u64 {
+        self.value.fetch_add(1, Ordering::Relaxed)
+    }
+}
+
```

Comment:
Mind adding a `Gauge` as well? From previous experience, counter and gauge should get us pretty far. Just a simple get/set (vs. get/inc in the counter). For the gauge, it'd be nice if it's typed so we can have string gauges, u64 gauges, and so on.